### PR TITLE
Bicaws 2125 Build docker image for UI

### PR DESCRIPTION
**Pull request by C-gyorfi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+FROM ${BUILD_IMAGE} as app_builder
+
+LABEL maintainer="CJSE"
+
+WORKDIR /src/next-ui
+
+COPY ./package*.json ./
+COPY ./scripts/ ./scripts/
+
+RUN npm install
```
Comment (jaskaransarkaria) [inline Dockerfile:14]: you can use `npm ci` here https://docs.npmjs.com/cli/v8/commands/npm-ci

**Review by tomvaughan77 — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    groupadd nodejs && \
+    usermod -a -G nodejs nextjs
+
+ENV NEXT_TELEMETRY_DISABLED=1
+ENV NODE_ENV=production
+
+WORKDIR /app
+COPY ./package*.json ./
+
+RUN npm install --production
```
Comment (jaskaransarkaria) [inline Dockerfile:35]: I think we only need one npm install, because the the node_modules are copied over on line 16

**Review by monotypical — APPROVED** [review]

**Review by tomvaughan77 — APPROVED** [review]

File: docker/conf/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  access_log              /dev/stdout combined;
+  gzip on;
+  include                 /etc/nginx/conf.d/*.conf;
+  include                 /etc/nginx/sites-enabled/*;
+  
+  log_format json_combined escape=json '{'
+      '"@timestamp": "$time_iso8601", '
+      '"message": "$remote_addr - $remote_user [$time_local] \\\"$request\\\" $status $body_bytes_sent \\\"$http_referer\\\" \\\"$http_user_agent\\\"", '
+      '"tags": ["nginx_access"], '
+      '"realip": ""$remote_addr", '
```
Comment (daviesjamie) [inline docker/conf/nginx.conf:31]: There's a typo here - `""$remote_addr"` is invalid JSON, it should be `"$remote_addr"`

I'd be interested in knowing where you copied this from, I thought I'd fixed all of these typos...

File: Dockerfile (hunk 2 above)
Comment (C-gyorfi) [reply Dockerfile]: Resolved in [7aadc3f](https://github.com/ministryofjustice/bichard7-next-ui/pull/5/commits/7aadc3f357b1ee4fe2524a56699df31d3117abed)

File: Dockerfile (hunk 1 above)
Comment (C-gyorfi) [reply Dockerfile]: Resolved in [7aadc3f](https://github.com/ministryofjustice/bichard7-next-ui/pull/5/commits/7aadc3f357b1ee4fe2524a56699df31d3117abed)

File: docker/conf/nginx.conf (hunk 3 above)
Comment (C-gyorfi) [reply docker/conf/nginx.conf]: Resolved in [7aadc3f](https://github.com/ministryofjustice/bichard7-next-ui/pull/5/commits/7aadc3f357b1ee4fe2524a56699df31d3117abed)

**Pull request merged**